webapp/login/oauth_views.py

```diff
@@ -0,0 +1,80 @@
+import os
+
+import flask
+from webapp.decorators import login_required
+from webapp.api.requests import Session
+from urllib.parse import urlencode
+from werkzeug.exceptions import BadRequest
+
+
+oauth = flask.Blueprint(
+    "oauth", __name__, template_folder="/templates", static_folder="/static",
+)
+
+session = Session()
```

Comment:
@nottrobin I am now not sure about this

Reply:
Do you mean, should you be simply using `Session()` on line 66? Is this the only place this session is used? If that's the case, that would seem reasonable to me.

Reply:
Okay, Yes it is the only place so I have moved it there now, but I was wondering because of the problem we had with github when we instantiated here in which users use the same instance. If at the performance level, in this case, it is better or not

Reply:
I can't imagine that the instantiation of a session object is a particularly expensive operation, so doing it once per request shouldn't be a problem. So in this case I think moving it into the view function is definitely best.